src/cmake/modules/GUDHI_third_party_libraries.cmake

```diff
@@ -134,23 +134,25 @@ add_definitions( -DBOOST_SYSTEM_NO_DEPRECATED )
 
 if (WITH_GUDHI_PYTHON)
   # Find the correct Python interpreter.
-  # Can be set with -DPYTHON_EXECUTABLE=/usr/bin/python3 or -DPython_ADDITIONAL_VERSIONS=3 for instance.
-  find_package( PythonInterp )
-  
+  # Can be set with -DPython_EXECUTABLE=/usr/bin/python3 for instance.
+  # CMP0094 OLD finds system python interpreter instead of conda even when using '-DCMAKE_PREFIX_PATH=$CONDA_PREFIX'
+  cmake_policy(SET CMP0094 NEW)
+  find_package( Python COMPONENTS Interpreter Development.Module NumPy)
```

Comment:
It looks like `Development.Module` (as opposed to plain `Development` which also includes `Development.Embed`) requires CMake 3.18, so we should either require that, or say `Development` (and put a comment next to it to remind us to update it in a couple years), or remove it.

Reply:
`find_package( Python COMPONENTS Interpreter Development NumPy)` and a comment on [630094b](https://github.com/GUDHI/gudhi-devel/pull/1112/commits/630094bfea2811de28292a81cfa9459a02885d39)